### PR TITLE
Add DotNetToolPackageRule to NGPV

### PR DESCRIPTION
**Pull request by natemcmaster**
Add support for a new package type - global CLI tools.

Also, fixed NullReferenceException's when running NGPV on the CLI tool packages

### CODE REVIEW

**Comment by natemcmaster** [comment]
FYI DotnetGlobalTool is just a placeholder name. The spec is unclear on what the `packageType` value should be for global CLI tools.

File: modules/NuGetPackageVerifier/console/PackageIssueFactory.cs [hunk 1]
```diff
@@ -39,7 +66,7 @@ public static PackageVerifierIssue AssemblyMissingServicingAttribute(string asse
                 PackageIssueLevel.Error);
         }
 
-        public static PackageVerifierIssue AssemblyInformationalVersionDoesNotMatchPackageVersion(string assemblyPath, NuGetVersion assemblyInformationalVersion, NuGetVersion packageVersion, string packageId)
+        public static PackageVerifierIssue AssemblyInformationalVersionDoesNotMatchPackageVersion(string assemblyPath, string assemblyInformationalVersion, NuGetVersion packageVersion, string packageId)
```
Comment (ryanbrandenburg) [inline modules/NuGetPackageVerifier/console/PackageIssueFactory.cs]: Curious why we went from more to less specific?

File: modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                if (entryPoint?.Value == null)
+                {
+                    yield return PackageIssueFactory.DotNetToolMalformedManifest("Missing EntryPoint");
+                    continue;
+                }
+
+                var runner = command.Attribute("Runner");
+                if (runner?.Value != "dotnet")
+                {
+                    continue;
```
Comment (ryanbrandenburg) [inline modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs]: Should there be an error here? Should we only be doing the above checks on dotnet tools or on everything?

File: modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                }
+
+                // In the initial implementation of global dools, EntryPoint is just a filename, not a full path inside the package.
```
Comment (ryanbrandenburg) [inline modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs]: "global dools"

File: modules/NuGetPackageVerifier/console/PackageIssueFactory.cs (hunk 1 above)
Comment (natemcmaster) [reply modules/NuGetPackageVerifier/console/PackageIssueFactory.cs]: For when NuGetVersion.TryParse fails.

File: modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs (hunk 2 above)
Comment (natemcmaster) [reply modules/NuGetPackageVerifier/console/Rules/DotNetToolPackageRule.cs]: The goal was to skip further checks of Runner != "dotnet". Runner is not a required attribute on commands.

**Pull request merged**